examples/ai-ml/ray/main.tf

```diff
@@ -0,0 +1,365 @@
+
+  server_side_encryption_configuration = {
+    rule = {
+      apply_server_side_encryption_by_default = {
+        kms_master_key_id = aws_kms_key.objects.arn
+        sse_algorithm     = "aws:kms"
+      }
+    }
+  }
+}
+
+data "aws_iam_policy_document" "irsa_policy" {
+  statement {
+    actions   = ["s3:ListBucket"]
+    resources = ["${module.s3_bucket.s3_bucket_arn}"]
+  }
+  statement {
+    actions   = ["s3:*Object"]
+    resources = ["${module.s3_bucket.s3_bucket_arn}/*"]
+  }
+  statement {
+    actions = [
+      "ssm:PutParameter",
+      "ssm:DeleteParameter",
+      "ssm:GetParameterHistory",
+      "ssm:GetParametersByPath",
+      "ssm:GetParameters",
+      "ssm:GetParameter",
+      "ssm:DeleteParameters",
+      "ssm:DescribeParameters"
+    ]
+    resources = ["*"]
+  }
+}
+
+resource "aws_iam_policy" "irsa_policy" {
+  description = "IAM Policy for IRSA"
+  name_prefix = substr("${module.eks_blueprints.eks_cluster_id}-${local.namespace}-access", 0, 127)
+  policy      = data.aws_iam_policy_document.irsa_policy.json
+}
+
+module "cluster_irsa" {
+  source                     = "../../../modules/irsa"
+  kubernetes_namespace       = local.namespace
+  kubernetes_service_account = "${local.namespace}-sa"
+  irsa_iam_policies          = [aws_iam_policy.irsa_policy.arn]
+  eks_cluster_id             = module.eks_blueprints.eks_cluster_id
+  eks_oidc_provider_arn      = module.eks_blueprints.eks_oidc_provider_arn
+
+  depends_on = [module.s3_bucket]
+}
+
+
+data "kubectl_path_documents" "docs" {
+  pattern = "${path.module}/ray-clusters/*.yaml"
+  vars = {
+    namespace       = local.namespace
+    hostname        = var.eks_cluster_domain == null ? "" : var.eks_cluster_domain
+    account_id      = data.aws_caller_identity.current.account_id
+    region          = local.region
+    service_account = "${local.namespace}-sa"
+  }
+}
+
+# Ugly Terraform issue:
+# See https://github.com/gavinbunney/terraform-provider-kubectl/issues/58#issuecomment-718174177
+data "kubectl_path_documents" "count_hack" {
+  pattern = "${path.module}/ray-clusters/*.yaml"
+  vars = {
+    namespace       = ""
+    hostname        = ""
+    account_id      = ""
+    region          = ""
+    service_account = ""
+  }
+}
+
+resource "kubectl_manifest" "cluster_provisioner" {
+  count     = length(data.kubectl_path_documents.count_hack.documents)
+  yaml_body = element(data.kubectl_path_documents.docs.documents, count.index)
+
+  depends_on = [
+    module.cluster_irsa,
+    module.eks_blueprints_kubernetes_addons
+  ]
+}
+
+
+#---------------------------------------------------------------
+# Monitoring
+#---------------------------------------------------------------
+data "kubectl_path_documents" "prometheus" {
+  pattern = "${path.module}/monitoring/monitor.yaml"
+  vars = {
+    namespace = local.namespace
+  }
+}
+
+resource "kubectl_manifest" "prometheus" {
+  yaml_body = data.kubectl_path_documents.prometheus.documents[0]
```

Comment:
similar to above - its just one file so you could do a `templatefile()` here and use just one resource